### PR TITLE
NISQ-Analyzer Plugin

### PR DESCRIPTION
**Pull request by cloudsftp**
Add necessary containers for the NISQ-Analyzer plugin

### CODE REVIEW

**Comment by cloudsftp** [comment]
Going to update the user documentation in this PR as well

File: docker-compose.yml [hunk 1]
```diff
@@ -65,13 +70,198 @@ services:
...
+    image: rigetti/quilc
+    profiles: ["nisq"]
+    ports:
+      - "5017:5017"
+    command: -S -p 5017
+    networks:
+      - default
+
+  nisq-analyzer-db:
+    image: postgres
```
Comment (buehlefs) [inline docker-compose.yml:241]: `profiles` is missing

**Review by buehlefs — APPROVED** [review]

File: readme.md [hunk 2]
```diff
@@ -9,6 +9,8 @@ The user documentation for QHAna can be viewed on [qhana.readthedocs.io](https:/
 
 `docker-compose up`
 
+If you want to use the NISQ-Analyzer plugin, use `docker-compose --profile nisq up`.
```
Comment (salmma) [inline readme.md:12]: add a link to the nisq analyzer repo

**Review by salmma — CHANGES_REQUESTED** [review]
Comments regarding quickstart.md
- "The inputs can be validated with the "validate" button." - please set a "valid" when the input for a plugin is correct - otherwise the user doesn't know if it worked
- "The title of the output directly links to the produced output data detaisl page. The page contains adownload link and a preview." fix typos.
- "Once the results are available the UI will show the outputs generated by the plugin in the open timeline step." - all screenshots afterwards are old (the menu bar is changed) please update - otherwise it is confusing for the user

Bildschirmfoto 2022-10-21 um 08 51 44
How can the user now start the analysis of an implementation after start up? It is not clear how to add an implementation - please add documentation.

**Review by salmma — CHANGES_REQUESTED** [review]
please also motivate the nature of templates: for what do they exist, how are they defined, who can define them, etc.

**Comment by cloudsftp** [comment]
> "The inputs can be validated with the "validate" button." - please set a "valid" when the input for a plugin is correct - otherwise the user doesn't know if it worked

I did not understand this point.

**Comment by salmma** [comment]
> > "The inputs can be validated with the "validate" button." - please set a "valid" when the input for a plugin is correct - otherwise the user doesn't know if it worked
> 
> I did not understand this point.

In the UI, when the user clicks on the "validate" button there is no hint that the input is valid (eg green small text or a green checkmark next to it), instead nothing happens

**Comment by buehlefs** [comment]
Changing the validate button behaviour is way out of scope for this PR. I've opened an Issue in the plugin runner repo for this <https://github.com/UST-QuAntiL/qhana-plugin-runner/issues/46>
@cloudsftp If you want to fix it, then you can, but I think that this is not part of your project (meaning that I am responsible to fix that).

**Comment by salmma** [comment]
> Changing the validate button behaviour is way out of scope for this PR. I've opened an Issue in the plugin runner repo for this [UST-QuAntiL/qhana-plugin-runner#46](https://github.com/UST-QuAntiL/qhana-plugin-runner/issues/46)
> @cloudsftp If you want to fix it, then you can, but I think that this is not part of your project (meaning that I am responsible to fix that).

the main thing is to fix it 🙃

**Review by salmma — CHANGES_REQUESTED** [review]
only one comment, rest looks good :)

**Review by salmma — APPROVED** [review]
LG

**Pull request merged**